pkg/client/transaction.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/codenotary/immudb/pkg/client/errors"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+type TxOptions struct {
+	TxMode schema.TxMode
+}
+
+type Tx interface {
+	Commit(ctx context.Context) (*schema.CommittedSQLTx, error)
+	Rollback(ctx context.Context) error
+
+	SQLExec(ctx context.Context, sql string, params map[string]interface{}) error
+	SQLQuery(ctx context.Context, sql string, params map[string]interface{}) (*schema.SQLQueryResult, error)
+}
+
+type tx struct {
+	ic            *immuClient
+	transactionID string
+}
+
+func (c *tx) Commit(ctx context.Context) (*schema.CommittedSQLTx, error) {
+	cmtx, err := c.ic.ServiceClient.Commit(ctx, new(empty.Empty))
+	return cmtx, errors.FromError(err)
+}
+
+func (c *tx) Rollback(ctx context.Context) error {
+	_, err := c.ic.ServiceClient.Rollback(ctx, new(empty.Empty))
+	return errors.FromError(err)
+}
+
+func (c *immuClient) BeginTx(ctx context.Context, options *TxOptions) (Tx, error) {
+
+	if options.TxMode == schema.TxMode_WRITE_ONLY {
+		// only in key-value mode, in sql we read catalog and write to it
```

Comment:
just to be sure, this validation should also be in the server

Reply:
already therehttps://github.com/codenotary/immudb/pull/1014/files#diff-6d3b6766c2d825705f2eb1e6fd1e3e84c44c8ab7be2fb28c7190d44247fd0851R73